alexherlanda/db8-events
# Updates Search UX and Implements Redux-Sagas 🔍

## Abstract 
Updates UI and  Front-end to handle search and refactor a little bit front-end strategy to handle side effects. UI is based on Youtube, searching behavior in Netflix and Spotify, 

## Purpose
Make React deal only with pure functions and use Redux and Redux-Sagas to support async requests

## Summary 
- Implements global request with interceptors to handle error regarding service globally 
- Implements redux-sagas to handle side effects
- Refactors client to use pure functions only
- Implements general search with `onChange` and `throttle` strategy 
- Hides filters just for the moment 

## How to test
- Run `yarn dev`
- Check that the server and client run
- Click the search icon 
- Enter a term for searching, check that the front makes petitions onChange but controlling that only one exists on a 1500ms period 
- Click the back icon 
- Check the main bar restores to normal


## An image tells more than 1000 words:
![image](https://user-images.githubusercontent.com/43116659/88489683-631c6400-cf5b-11ea-8771-7c9174971a95.png)
![image](https://user-images.githubusercontent.com/43116659/88489715-95c65c80-cf5b-11ea-8687-b0b102ee7ba7.png)
![image](https://user-images.githubusercontent.com/43116659/88489723-a971c300-cf5b-11ea-87b8-e3105d25d210.png)



## Next steps 
- Add the filters that Julian have already built on Airbnb based UI
- Implement more media queries
- Add some animations
